shotover-proxy/src/message/mod.rs

```diff
@@ -232,6 +241,56 @@ impl Message {
         self.invalidate_cache();
     }
 
+    /// Get metadata for this `Message`
+    fn metadata(&mut self) -> Result<Metadata> {
+        let inner = self.inner.take().unwrap();
+        match inner {
+            MessageInner::RawBytes {
+                bytes,
+                message_type,
+            } => match message_type {
+                MessageType::Cassandra => Ok(Metadata::Cassandra(cassandra::metadata(bytes)?)),
+                MessageType::Redis => Ok(Metadata::Redis),
+                MessageType::None => Ok(Metadata::Unknown),
+            },
+            MessageInner::Parsed { frame, .. } | MessageInner::Modified { frame } => match frame {
+                Frame::Cassandra(frame) => Ok(Metadata::Cassandra(frame.metadata())),
+                Frame::Redis(_) => Ok(Metadata::Redis),
+                Frame::None => Ok(Metadata::Unknown),
+            },
+        }
+    }
+
+    /// Set this `Message` to a backpressure response
+    pub fn set_backpressure(&mut self) -> Result<()> {
+        let metadata = self.metadata()?;
+
+        *self = Message::from_frame(match metadata {
+            Metadata::Cassandra(metadata) => {
+                let body = CassandraOperation::Error(ErrorBody {
+                    error_code: 0x1001,
+                    message: "".into(),
+                    additional_info: AdditionalErrorInfo::Overloaded,
+                });
+
+                Frame::Cassandra(CassandraFrame {
+                    version: metadata.version,
+                    stream_id: metadata.stream_id,
+                    tracing_id: metadata.tracing_id,
+                    warnings: metadata.warnings,
```

Comment:
I'm not sure it makes sense to return the warnings back to the client?
If we can remove it here then we can also remove it from metadata.

Reply:
![image](https://user-images.githubusercontent.com/5120858/156467159-0c3fdc11-810c-4c96-be9d-89353b1ec1f2.png)
That sounds like:
1. the client wont send warnings anyway
2. even if it did it wouldnt make sense to send those warnings back.

Reply:
Yep, good spot. I'll fix that up.

Reply:
Maybe we should leave it in the metadata struct since it technically is part of a Cassandra message's metadata, but remove it when we are setting a message to backpressure?

Reply:
yeah thats fine for now.

Reply:
Had to remove it because it's unread, but I added a comment to the struct.